pkg/tsdb/cloudwatch/log_actions.go

```diff
@@ -33,12 +32,11 @@ func (e *CloudWatchExecutor) executeLogActions(ctx context.Context, queryContext
 			// the query response is in, there does not seem to be a way to tell
 			// by the response alone if/how the results should be grouped.
 			// Because of this, if the frontend sees that a "stats ... by ..." query is being made
-			// the "groupResults" parameter is sent along with the query to the backend so that we
+			// the "statsGroups" parameter is sent along with the query to the backend so that we
 			// can correctly group the CloudWatch logs response.
-			if query.Model.Get("groupResults").MustBool() && len(dataframe.Fields) > 0 {
-				groupingFields := findGroupingFields(dataframe.Fields)
-
-				groupedFrames, err := groupResults(dataframe, groupingFields)
+			statsGroups := query.Model.Get("statsGroups").MustStringArray()
```

Comment:
Su just to understand this, `query.Model.Get("statsGroups")` is filled in on the front end where we use the parser, right? What happens in case of alerting? Do we store this somewhere with the query? Not sure right now what the alerting data model is.

Reply:
The parsed stats groups are stored as part of the query, so when the query gets executed on the backend as part of the alert system, it knows to group the results correctly.
Ideally we would have that stats group parsing done on the backend for alerts rather than storing them as part of the query, but without access to a parser generated for go it wouldn't be too easy